resources/js/chat/main-view.tsx

```diff
@@ -28,6 +29,17 @@ export default class MainView extends React.Component<Record<string, never>> {
         }
       }),
     );
+
+    disposeOnUnmount(
+      this,
+      autorun(() => {
+        if (core.dataStore.chatState.isAddingChannel) {
+          showImmediateLoadingOverlay();
+        } else {
+          hideLoadingOverlay();
```

Comment:
I know there is an issue when going back/forward and then returning, but I'm not planning to fixing it in this PR.